cmd/main_test.go

```diff
@@ -301,3 +307,138 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("Should return a WorkspaceList with test-tenant workspace and both namespaces in it", func() {
+			actualWorkspaces, err := getWorkspacesWithAccess(e, c, allNamespaces)
+			Expect(actualWorkspaces.Items).To(Equal(expectedWorkspaces))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetWorkspacesWithAccess")
+		})
+
+		AfterEach(func() {
+			allNamespaces = []k8sapi.Namespace{}
+			expectedWorkspaces = []crt.Workspace{}
+		})
+	})
+
+	Context("When workspace with only test-tenant namespace has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"ws-test-tenant-3", "ws-test-tenant-4"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			createRole(k8sClient, "ws-test-tenant-3", "ws-namespace-access-3", []string{"create", "list", "watch", "delete"})
+			createRoleBinding(k8sClient, "ws-namespace-access-user-binding-3", "ws-test-tenant-3", "user1@konflux.dev", "ws-namespace-access-3")
+			expectedWorkspaces = []crt.Workspace{
+				{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Workspace",
+						APIVersion: gv,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "ws-test-tenant-3",
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "ws-test-tenant-3",
+								Type: "default",
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("Should return a WorkspaceList with test-tenant workspace and only test-tenant namespace in it", func() {
+			actualWorkspaces, err := getWorkspacesWithAccess(e, c, allNamespaces)
+			Expect(actualWorkspaces.Items).To(Equal(expectedWorkspaces))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetWorkspacesWithAccess")
+		})
+
+		AfterEach(func() {
+			allNamespaces = []k8sapi.Namespace{}
+			expectedWorkspaces = []crt.Workspace{}
+		})
+	})	
+
+	Context("When no workspaces has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"ws-test-tenant-4", "ws-test-tenant-5"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			expectedWorkspaces = []crt.Workspace{}
+		})
+
+		It("Should return a empty WorkspaceList", func() {
+			actualWorkspaces, err := getWorkspacesWithAccess(e, c, allNamespaces)
+			Expect(actualWorkspaces.Items).To(Equal(expectedWorkspaces))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetWorkspacesWithAccess")
+		})
+
+		AfterEach(func() {
+			allNamespaces = []k8sapi.Namespace{}
```

Comment:
Those are not the variables you need to clean, 
what we need to clean are the actual resources.

for example:
in the test you ran the following- 
```
createNamespace(k8sClient, name)
createRole(k8sClient, name, "namespace-access", []string{"create", "list", "watch", "delete"})
createRoleBinding(k8sClient, "namespace-access-user-binding", name, "user1@konflux.dev", "namespace-access")
```

those resources may stay available on the envtest cluster while the other tests are running, as we use the same env. we should remove all resources we created.